### PR TITLE
Update GoogleSheetMapper to programatically load more records

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Describe this PR

Updates to the LegalDisrepair integration.

To make sure all legalDisrepair records are loaded, I have added a mechanism the loads the first 500, then if the last 10 rows arent empty, loads the next 500. This approach means we can load the least number of records, while ensuring all records are loaded.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### CODE REVIEW

File: RepairsApi.Tests/V2/Services/GoogleSheets/LegalDisrepairMapperTests.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RepairsApi.Tests.V2.Services.GoogleSheets
+{
+    public class LegalDisrepairMapperTests
+    {
+        private LegalDisrepairMapper _classUnderTest;
+
+        private Mock<IGoogleSheetGateway<LegalDisrepairCredentials>> _mockGateway;
```
Comment (Asura5) [inline RepairsApi.Tests/V2/Services/GoogleSheets/LegalDisrepairMapperTests.cs:22]: Reusability: Does this have to be LegalDisrepairCredentials, or can it be GoogleSheetCredentials

File: RepairsApi/Startup.cs [hunk 2]
```diff
@@ -231,9 +231,16 @@ private void ConfigureGoogleSheetServices(IServiceCollection services)
             var options = new GoogleSheetsOptions();
             Configuration.Bind(nameof(GoogleSheetsOptions), options);
 
-            services.AddTransient<IGoogleSheetsMapper<LegalDisrepairData>>(sp =>
-                new LegalDisrepairMapper(ConfigureSheetsService(options.LegalDisrepairCredentials), options.LegalDisrepairSpreadsheetId, new LoggerFactory()));
+            var legalDisrepairCredentials = new LegalDisrepairCredentials(options.LegalDisrepairSpreadsheetId);
 
+            // Add GoogleSheetGateway for each integration
+            services.AddTransient<IGoogleSheetGateway<LegalDisrepairCredentials>>(sp =>
+                new GoogleSheetGateway<LegalDisrepairCredentials>(ConfigureSheetsService(options.LegalDisrepairCredentials), legalDisrepairCredentials, new LoggerFactory()));
```
Comment (Asura5) [inline RepairsApi/Startup.cs:238]: Code smell - line 234+238 could do with a refactor - you are using options.LegalDisrepairSpreadsheetId both directly and indirectly

**Review by Asura5 — CHANGES_REQUESTED** [review]
See comments

File: RepairsApi.Tests/V2/Services/GoogleSheets/LegalDisrepairMapperTests.cs (hunk 1 above)
Comment (LBHCallumM) [reply RepairsApi.Tests/V2/Services/GoogleSheets/LegalDisrepairMapperTests.cs:22]: The intention was to make dependency injection easier. Lets discuss this in tomorrows dropin session .

File: RepairsApi/Startup.cs (hunk 2 above)
Comment (LBHCallumM) [reply RepairsApi/Startup.cs]: > See comments

I have removed the `IGoogleSheetCredentials` interface from GoogleSheetGateway.

**Pull request closed without merging**